specreduce/extract.py

```diff
@@ -497,50 +649,79 @@ def __call__(self, image=None, trace_object=None,
-                                       mean=coadd.argmax(), stddev=2)
-
-        # Fit extraction kernel to column's finite values with combined model
-        # (must exclude masked indices manually; LevMarLSQFitter does not)
-        ext_prof = gauss_prof + bkgrd_prof
-        fitter = fitting.LevMarLSQFitter()
-        fit_ext_kernel = fitter(ext_prof,
-                                xd_pixels[~row_mask], coadd[~row_mask])
+                crossdisp_axis=crossdisp_axis)
+
+        if self.spatial_profile == 'gaussian':
+
+            # fit profile to average (mean) profile along crossdisp axis
+            fit_ext_kernel = self._fit_gaussian_spatial_profile(img,
+                                                                disp_axis,
+                                                                crossdisp_axis,
+                                                                or_mask,
+                                                                bkgrd_prof)
+
+            # this is just creating an array of the trace to shift the mean
+            # when iterating over each wavelength. this needs to be fixed in the
+            # future to actually account for the trace shape in a non-flat trace
+            # (or possibly omitted all togehter as it might be redundant if
+            # _align_along_trace is correcting this already)
+            if isinstance(trace_object, FlatTrace):
+                mean_init_guess = trace_object.trace
+            else:
+                mean_init_guess = np.broadcast_to(
+                    img.shape[crossdisp_axis] // 2, img.shape[disp_axis]
+                )
+
+        else:  # self_profile
+            # for now, bkgrnd_prof must be None because a compound model can't
+            # be created with a interpolator + model. i think there is a way
+            # around this, but will follow up later
+            if bkgrd_prof is not None:
+                raise ValueError('When `spatial_profile`is `self_profile`,'
+                                 '`bkgrnd_prof` must be None. Background should'
+                                 ' be fit and subtracted from `img` beforehand.')
+            # make sure n_bins doesnt exceed the number of (for now) finite
+            # columns. update this when masking is fixed.
+            n_finite_cols = np.logical_or.reduce(or_mask, axis=crossdisp_axis)
+            n_finite_cols = np.count_nonzero(n_finite_cols.astype(int) == 0)
+
+            # determine interpolation degree from input and make
+            if isinstance(interp_degree_self_profile, int):
+                kx = ky = interp_degree_self_profile
+            else:  # if input is tuple of ints
+                kx, ky = interp_degree_self_profile
+
+            if n_bins_self_profile >= n_finite_cols:
+
+                raise ValueError(f'`n_bins_self_profile` ({n_bins_self_profile}) '
+                                 'should not exceed the number of fully-finite '
```

Comment:
This wording could be confusing for the user when `n_bins_self_profile == n_finite_cols`

```suggestion
                                 'must be less than the number of fully-finite '
```